### PR TITLE
Simplify ST-Link and JLink USB code

### PR DESCRIPTION
**Pull request by zyp**
## Detailed description

This is the followup to #1478, bringing the same simplification to ST-Link and JLink USB interface drivers.

Not yet tested on hardware.

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds for hardware native (`make PROBE_HOST=native`)
* [x] It builds as BMDA (`make PROBE_HOST=hosted`)
* [x] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

File: src/platforms/hosted/stlinkv2.c
Comment (dragonmux) [inline src/platforms/hosted/stlinkv2.c:417]: Does this not still need to return false when triggered?

File: src/platforms/hosted/jlink.c
Comment (dragonmux) [inline src/platforms/hosted/jlink.c:177]: Does this not need to return false when triggered?

File: src/platforms/hosted/jlink.c
Comment (dragonmux) [inline src/platforms/hosted/jlink.c:178]: The {}'s are now excess here, please remove them

**Review by dragonmux — CHANGES_REQUESTED** [review]
This is a fantastic simplification! We'll test this after getting some sleep. Many thanks zyp.

The only items we've spotted in review are some possibly missing `return false;` and one now excessive set of braces. With those fixed and after a successful test, we'll be happy to merge this.

**Review by dragonmux — APPROVED** [review]
LGTM, we'll merge this after getting it tested.

**Comment by dragonmux** [comment]
ST-Link tested working

**Review by perigoso — APPROVED** [review]
Tested with a JLink V8 and a JLink OB (BRD2207A)

There seem to be issues with the JLink code, getting libusb errors with both but looks unrelated to the changeset as behavior is unchanged with the base commit
I was able to connect and run *some* commands with this

**Comment by dragonmux** [comment]
Excellent, we'll merge this then

**Pull request merged**
